Total rewrite (#35)
martinduke/congestion-control-charter: congestion-control-charter.txt

Since RFC 5033 was published, many conditions have changed. Congestion control
algorithm proponents now often have the opportunity to test and deploy at scale
without IETF review. The set of protocols using these algorithms has spread
beyond TCP and SCTP to include DCCP, QUIC, and beyond. There is more interest
in specialized use cases such as data centers and real-time protocols. Finally,
the community has gained much more experience with indications of congestion
beyond packet loss.

The Congestion working group will review some of the impediments to congestion
control work occurring in the IETF, and can generalize transport in this area
from TCP to all of the relevant transport protocols. This will inform a revision
of RFC 5033 that encourages IETF review of congestion control proposals and
standardization of mature congestion control algorithms.

The congestion control expertise in the working group would also make it a
natural venue to take on other work related to indications of congestion (such
as delay), queueing algorithms, rate pacing, interaction with other layers,
etc. In particular, it could address congestion control algorithms mature enough

> for standardization. The working group is intended to be a home for such work,

Reviewer: I do think it's worth to clarify the "mature enough for standardization" with the two criteria we had in the old version, just to make it clearer for everyone what the distinction is between this WG and ICCRG/IRTF work: 
```
(1) empirical evidence of safety (for example - avoiding congestion collapse) and 
(2) stated intent to deploy by major transport implementations.
```

Related to Éric's comment, I wonder if we should mention experimentation as an example of empirical evidence, but not the only one -- I'd say having an algorithm deployed in production for a certain period of time can also serve as empirical evidence, even in the absence of a published formal experimentation methodology. (That's why I don't want to simply change "empirical" to "experimental".)

should it arise, if not suitable for the IRTF. It is chartered to consider
proposals in this space, and given consensus to adopt, seek an immediate
recharter regardless of the status of the initial deliverable.

> Upon completion of the deliverable, Congestion WG will close in the absence of

Reviewer: The following sounds a bit smoother to me (as it'll be true whether or not we get additional milestones after 5033bis):

```suggestion
Upon completion of its deliverables, the Congestion WG will close in the absence of

```

﻿Congestion Working Group Charter

RFC 5033 describes a Best Current Practice to evaluate new congestion control
algorithms as Experimental or Proposed Standard RFCs. TCP was the dominant
consumer of this work, and proposals were typically discussed in research

> groups (for example, Internet Congestion Control Research Group - ICCRG).

Reviewer 2: ```suggestion
groups (for example, the Internet Congestion Control Research Group - ICCRG).

```

Since RFC 5033 was published, many conditions have changed. Congestion control
algorithm proponents now often have the opportunity to test and deploy at scale
without IETF review. The set of protocols using these algorithms has spread
beyond TCP and SCTP to include DCCP, QUIC, and beyond. There is more interest
in specialized use cases such as data centers and real-time protocols. Finally,
the community has gained much more experience with indications of congestion
beyond packet loss.

The Congestion working group will review some of the impediments to congestion

> control work occurring in the IETF, and can generalize transport in this area

Reviewer 2: ```suggestion
control work occurring in the IETF and can generalize transports

```

from TCP to all of the relevant transport protocols. This will inform a revision
of RFC 5033 that encourages IETF review of congestion control proposals and
standardization of mature congestion control algorithms.

The congestion control expertise in the working group would also make it a
natural venue to take on other work related to indications of congestion (such
as delay), queueing algorithms, rate pacing, interaction with other layers,
etc. In particular, it could address congestion control algorithms with
empirical evidence of safety and stated intent to deploy by major

> implementation. The working group is intended to be a home for such work, should

Reviewer 2: ```suggestion
implementations. The working group is intended to be a home for such work, should

```

RFC 5033 describes a Best Current
Practice to evaluate new congestion control algorithms as Experimental or
Proposed Standard RFCs. TCP was the dominant
consumer of this work, and proposals were typically discussed in research groups (for example, Internet Congestion Control Research Group - ICCRG).

Since RFC 5033 was published, many conditions have changed. Congestion control algorithm proponents
now often have the opportunity to test and deploy at scale without IETF review. The
set of protocols using these algorithms has spread beyond TCP and SCTP to
include DCCP, QUIC, and beyond. There is more interest in specialized use cases
such as data centers and real-time protocols. Finally, the community has gained
much more experience with indications of congestion beyond packet loss.

The CONGestion RESponse and Signaling working group (CONGRESS) can review some
of the impediments to early congestion control work occurring in the IETF, and
can generalize transport in this area from TCP to all of the relevant transport
protocols. The congestion control expertise in the working group would also
make it a natural venue to take on other work related to indications of

> congestion, such as delay, and potentially queueing algorithms.

Reviewer 3: Seems like we still need something like this paragraph.

Reviewer 2: I think this goes to the point of https://github.com/martinduke/congestion-control-charter/pull/35#discussion_r1152729095, so let's either include or remove it based on the outcome there.

The congestion control expertise in the working group would also make it a
natural venue to take on other work related to indications of congestion (such
as delay), queueing algorithms, rate pacing, interaction with other layers,
etc. In particular, it could address congestion control algorithms with
empirical evidence of safety and stated intent to deploy by major
implementations. The working group is intended to be a home for such work, should
it arise, if not suitable for the IRTF. It is chartered to consider proposals in
this space, and given consensus to adopt, seek an immediate recharter regardless

> of the status of the initial deliverable.

Reviewer 3: These paragraph needs some more work I think. This is not clear to me what we are trying say clearly here. Like the meaning of "it could address" is not very clear. Will we be just discussing the feasibility of publishing focusing on safety and deployment and then recharter to actually publish something? or will also host the demonstration of algorithm efficiency of the particular algorithms which now usually hosted by ICCRG. 

To me one of the purpose of creating the group is to able to publish standard track congestion control algorithms. I think we need to be clear about that intention in the charter. Otherwise, we just do the -bis work and after the -bis work done we close or recharter to publish std track congestion control algorithm.

Reviewer 2: It had sounded like the ask was to do the latter: do the -bis work and then recharter, but I would also be okay if we wanted to include more of that additional context up front, would probably help to not be solely focused on the -bis and instead have some real world test cases for the procedures outlined there.

Reviewer: Agree with Eric. Given that we have a few months until IETF 117, perhaps there's an opportunity to see if we can get more concrete work items (on the horizon), and then add related text to the charter.
Most importantly, I would like the charter to be clear that we're interested in **solutions** (e.g., algorithms), and not limit ourselves to procedures, problem statements, or theoretical frameworks.

I'd be happy to land this PR now, and then iterate and add text related to the concrete work on the horizon, but be sure to not bring back the long list of ideas without concrete work on the horizon.

Reviewer 4: What does "if not suitable for IRTF" means? Does this wg only take solution which is deployed and tested in the wild?

Reviewer 2: Updated to be more clear about the initial litmus test: empirical evidence of safety and stated intent to deploy by major implementations. Deployment and testing in the wild is one way to provide empirical evidence of safety, yes.

Upon completion of its deliverables, the Congestion Control WG will close in 

> the absence of suitable proposals for further work.

Reviewer 3: Looks unnecessary to me as it would be natural to close the WG if is has nothing to work with.

Reviewer 2: This has gone in and out, that is the default state of things, but it seems okay to state it explicitly.

> ﻿Congestion Working Group Charter

Reviewer 5: ```suggestion
﻿Congestion Control Working Group Charter
```

> ﻿Congestion Working Group Charter

Reviewer: ```suggestion
﻿Congestion Control Working Group Charter (CCWG)

```

RFC 5033 describes a Best Current Practice to evaluate new congestion control
algorithms as Experimental or Proposed Standard RFCs. TCP was the dominant
consumer of this work, and proposals were typically discussed in research
groups (for example, the Internet Congestion Control Research Group - ICCRG).

> 

Reviewer 5: ```suggestion

```

Since RFC 5033 was published, many conditions have changed. Congestion control
algorithm proponents now often have the opportunity to test and deploy at scale
without IETF review. The set of protocols using these algorithms has spread
beyond TCP and SCTP to include DCCP, QUIC, and beyond. There is more interest
in specialized use cases such as data centers and real-time protocols. Finally,
the community has gained much more experience with indications of congestion
beyond packet loss.

> The Congestion working group will review some of the impediments to congestion

Reviewer 5: ```suggestion
The Congestion Control Working Group will review some of the impediments to congestion
```

> The Congestion working group will review some of the impediments to congestion

Reviewer: ```suggestion
The Congestion Control Working Group will review some of the impediments to congestion

```

control work occurring in the IETF and can generalize transports

> 

Reviewer 5: ```suggestion

```

from TCP to all of the relevant transport protocols. This will inform a revision
of RFC 5033 that encourages IETF review of congestion control proposals and
standardization of mature congestion control algorithms.

The congestion control expertise in the working group would also make it a
natural venue to take on other work related to indications of congestion (such
as delay), queueing algorithms, rate pacing, interaction with other layers,
etc. In particular, it could address congestion control algorithms with
empirical evidence of safety and stated intent to deploy by major
implementations. The working group is intended to be a home for such work, should
it arise, if not suitable for the IRTF. It is chartered to consider proposals in
this space, and given consensus to adopt, seek an immediate recharter regardless
of the status of the initial deliverable.

> Upon completion of its deliverables, the Congestion WG will close in the absence

Reviewer 5: ```suggestion
Upon completion of its deliverables, the Congestion Control WG will close in the absence
```

> Upon completion of its deliverables, the Congestion WG will close in the absence

Reviewer: ```suggestion
Upon completion of its deliverables, the Congestion Control WG will close in the absence

```
